### PR TITLE
Case list tile and persistent case tile fix

### PR DESCRIPTION
**Pull request by wpride**
Addresses http://manage.dimagi.com/default.asp?246931

The grid polyfill styles were overwriting each other when both a persistent case tile and a case list with tiles existed simultaneously. This splits out those concerns and makes the code (slightly) more functional. 

This still might be my least favorite thing I've ever written

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js [hunk 1]
```diff
@@ -137,19 +137,25 @@ FormplayerFrontend.module("Menus", function (Menus, FormplayerFrontend, Backbone
...
+                data: detailObject.details,
+                id: 0,
+            });
+            var numEntitiesPerRow = detailObject.numEntitiesPerRow || 1;
+            var numRows = detailObject.maxHeight;
+            var numColumns = detailObject.maxWidth;
+            var useUniformUnits = detailObject.useUniformUnits || false;
+            var caseTileStyles = Menus.Views.buildCaseTileStyles(detailObject.tiles, numRows, numColumns,
+                numEntitiesPerRow, useUniformUnits, 'persistent');
+            $("#persistent-cell-layout-style").html(caseTileStyles[0]).data("css-polyfilled", false);
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js]: can you add a comment as to what `0` and `1` mean?

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 2]
```diff
@@ -209,8 +200,8 @@ FormplayerFrontend.module("Menus.Views", function (Views, FormplayerFrontend, Ba
         templateHelpers: function () {
             var appId = Util.currentUrlToObject().appId;
             var data = this.options.model.get('data');
-            for (var i =0; i < data.length; i++) {
-                data[i] = data[i].replace(/\"/g, "'");
+            for (var i = 0; i < data.length; i++) {
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: 😁

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 3]
```diff
@@ -224,6 +215,11 @@ FormplayerFrontend.module("Menus.Views", function (Views, FormplayerFrontend, Ba
 
     Views.CaseTileView = Views.CaseView.extend({
         template: "#case-tile-view-item-template",
+        templateHelpers: function () {
+            var dict = Views.CaseTileView.__super__.templateHelpers.apply(this, arguments);
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: huh haven't seen `__super__` used before, do you know if it's well supported in browsers?

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 4]
```diff
@@ -224,6 +215,11 @@ FormplayerFrontend.module("Menus.Views", function (Views, FormplayerFrontend, Ba
...
+            dict['prefix'] = this.options.prefix;
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: nit: `dict.prefix = `

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 5]
```diff
@@ -296,19 +292,52 @@ FormplayerFrontend.module("Menus.Views", function (Views, FormplayerFrontend, Ba
...
+            if (caseTileStyles.length > 2) {
+                $("#list-cell-container-style").html(caseTileStyles[2]).data("css-polyfilled", false);
+            }
+
             $.getScript(gridPolyfillPath);
         },
 
+        childViewOptions: function () {
+            var dict = Views.CaseTileListView.__super__.childViewOptions.apply(this, arguments);
+            dict['prefix'] = 'list';
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: you can also add `prefix` as a property on the view itself. that way you can just do `this.prefix`

**Review by benrudolph — COMMENTED** [review]
Nice thanks for pushing on this. will let you respond to comments, nothing too major

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js (hunk 3 above)
Comment (wpride) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: I've been using this in this file for a while IE https://github.com/dimagi/commcare-hq/blob/3430355aa1e6ba40a38968dba3536dcb9aaef2c5/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js#L315-L315

So I guess relatively supported?

Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: 👍 sweet

**Pull request merged**